app/code/Magento/Sales/Console/Command/PaymentUpdateEncryptCommand.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Sales\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Command\Command;
+use Magento\Framework\Console\Cli;
+use Magento\Framework\Encryption\Encryptor;
+
+/**
+ * Command for updated encrypted credit card data to the latest cipher
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class PaymentUpdateEncryptCommand extends Command
+{
+    /** Command name */
+    const NAME = 'encryption:update:payment';
+    const LEGACY_PATTERN = '^[[:digit:]]+:[^%s]:.*$';
+
+    /**
+     * @var \Magento\Framework\Encryption\Encryptor
+     */
+    private $encryptor;
+
+    /**
+     * @var \Magento\Sales\Model\ResourceModel\Order\Payment
+     */
+    private $paymentResource;
+
+    /**
+     * @param \Magento\Framework\Encryption\Encryptor $encryptor
+     * @param \Magento\Sales\Model\ResourceModel\Order\Payment $paymentResource
+     */
+    public function __construct(
+        \Magento\Framework\Encryption\Encryptor $encryptor,
+        \Magento\Sales\Model\ResourceModel\Order\Payment $paymentResource
+    ) {
+        $this->encryptor = $encryptor;
+        $this->paymentResource = $paymentResource;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName(self::NAME)
+            ->setDescription(
+                'Re-encrypts encrypted credit card data with latest encryption cipher.'
+            );
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $connection = $this->paymentResource->getConnection();
```

Comment:
DB operations are too low-level to be in a CLI command (as it should be considered the same as a controller).
Could you please move DB logic to a model?